### PR TITLE
chore: migrate goal to v3/cli

### PR DESCRIPTION
**Pull request by Peter-MJ-Parker**
Changed the example command to use v3 and cli rather than publish plugin.

### CODE REVIEW

File: docs/guide/walkthrough/goal.md [hunk 1]
```diff
@@ -36,16 +36,28 @@ export class PingCommand extends Command {
...
+    plugins: [],
     description: 'Pong!',
     execute: (ctx, args) => {
         await ctx.reply('Pong!')
     }
 })
 ```
-Keep in mind the above example acts as both a slash command AND text command
+Keep in mind the above example acts as both a slash command AND text command.
+config is not required but is needed if you want to change perms or other properties.
```
Comment (SrIzan10) [inline docs/guide/walkthrough/goal.md:60]: change config to configuration

File: docs/guide/walkthrough/goal.md [hunk 2]
```diff
@@ -36,16 +36,28 @@ export class PingCommand extends Command {
...
+- `dmPermission` boolean Tells the bot weather the command can be used in BOT's DM's.
```
Comment (SrIzan10) [inline docs/guide/walkthrough/goal.md:61]: "the bot"

File: docs/guide/walkthrough/goal.md [hunk 3]
```diff
@@ -36,16 +36,28 @@ export class PingCommand extends Command {
...
+- `defaultMemberPermissions` Array<PermissionFlagsBits> to hide the command from specific members.
```
Comment (SrIzan10) [inline docs/guide/walkthrough/goal.md:62]: should we hardcode the djs docs link for `PermissionFlagsBits`?

File: docs/guide/walkthrough/goal.md [hunk 4]
```diff
@@ -36,16 +36,28 @@ export class PingCommand extends Command {
 ```
 ### Using @sern/handler
 ```ts title="commands/ping.ts" showLineNumbers
-import { commandModule, CommandType } from '@sern/handler'
-import { publish } from '../plugins';
+import { commandModule, CommandType } from '@sern/handler';
+import { PermissionFlagsBits } from 'discord.js';
+
+export const config = {
+  dmPermission: true || false, //only choose one!
```
Comment (SrIzan10) [inline docs/guide/walkthrough/goal.md:43]: i think you should replace it with "boolean", which will require the user to change it accordingly but will not show this bad and redundant code

File: docs/guide/walkthrough/goal.md [hunk 5]
```diff
@@ -36,16 +36,28 @@ export class PingCommand extends Command {
...
+	defaultMemberPermissions: [
+		PermissionFlagsBits.UseApplicationCommands
+	],
+  guildIds: ['id1', 'id2']
```
Comment (SrIzan10) [inline docs/guide/walkthrough/goal.md:47]: change to `guildId1`, `...2`, just to clear things up and read less code when copying the snippet

**Review by SrIzan10 — CHANGES_REQUESTED** [review]
take a look at these minor reviews!

File: docs/guide/walkthrough/goal.md (hunk 4 above)
Comment (Peter-MJ-Parker) [reply docs/guide/walkthrough/goal.md:43]: I done that below but forgot to place above. oops

**Comment by SrIzan10** [comment]
@sern-handler/developers can we merge this?

**Comment by jacoobes** [comment]
Don't think the reviews were changed yet,

**Comment by Peter-MJ-Parker** [comment]
The reviews have not been changed yet due to seren thinking we should pause the PR.

https://discord.com/channels/889026545715400705/1101189430938701854/1228119418475909183

**Comment by jacoobes** [comment]
sorry for the late response, go ahead and fix these up. we're going to migrate this to starlight as well

**Comment by SrIzan10** [comment]
we've migrated to starlight, so im sure this will be outdated by now.
im sorry. create another pull request if you still want to :wink:

**Pull request closed without merging**